pkg/odo/cli/component/watch.go

```diff
@@ -54,16 +53,13 @@ func NewWatchOptions() *WatchOptions {
 
 // Complete completes watch args
 func (wo *WatchOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
-	wo.client = genericclioptions.Client(cmd)
+	// Set the correct context
+	wo.Context = genericclioptions.NewContextCreatingAppIfNeeded(cmd)
```

Comment:
A bit lost here. Why do we need to create a context in `odo watch`? Isn't `odo watch` supposed to work if and only if user is working from a valid component directory or has specified CLI parameters that help deduce a valid context?

Reply:
context contains the `LocalConfigInfo` hence every command that requires a valid component dir would need context.